src/main/java/io/phdata/streamliner/schemadefiner/util/StreamlinerUtil.java

```diff
@@ -0,0 +1,210 @@
+
+        //TODO: keep the parameter sequence preserved. like name, environment, pipeline, source, destination
+        Yaml yaml = new Yaml(representer, opt);
+        PrintWriter writer = new PrintWriter(outputDir);
+        yaml.dump(outputConfig, writer);
+    }
+
+    private static void createDir(String outputDir) {
+        File f = new File(outputDir);
+        if (!f.exists()) {
+            log.debug("Creating directory: {}", outputDir);
+            f.mkdirs();
+        }
+    }
+
+    public static Configuration mapGlueCatalogToConfig(Configuration ingestConfig, StreamlinerCatalog catalog) {
+        Source source = ingestConfig.getSource();
+        Schema db = catalog.getSchemas().stream().filter(schema -> schema.getCatalogName().equalsIgnoreCase(source.getDatabase())).collect(Collectors.toList()).get(0);
+        List<Table> tableList = (List<Table>) catalog.getTables(db);
+        List<TableDefinition> tableDefinitions = null;
+        if (ingestConfig.getDestination().getType().equalsIgnoreCase("SNOWFLAKE")) {
+            tableDefinitions = SnowflakeMapper.mapAWSGlueTables(tableList, source.getUserDefinedTable());
+            ingestConfig.setTables(tableDefinitions);
+        } else if (ingestConfig.getDestination().getType().equalsIgnoreCase("HADOOP")) {
+            tableDefinitions = HadoopMapper.mapAWSGlueTables(tableList, source.getUserDefinedTable());
+            ingestConfig.setTables(tableDefinitions);
+        }
+        return ingestConfig;
+    }
+    public static void getHtmlOutput(Source source, String password, String path) throws Exception {
+        execute(source, password, TextOutputFormat.html, path, "schema.html");
+    }
+
+    public static void getErdOutput(Source source, String password, String path) throws Exception {
+        execute(source, password, DiagramOutputFormat.png, path, "schema.png");
+    }
+
+    private static void execute(Source source, String password, OutputFormat outputFormat, String path, String fileName) throws Exception {
+        createDir(path);
+        OutputOptions outputOptions = OutputOptionsBuilder.newOutputOptions(outputFormat, Paths.get(path + "/" + fileName));
+        SchemaCrawlerExecutable executable = new SchemaCrawlerExecutable("schema");
+        executable.setSchemaCrawlerOptions(getOptions(source));
+        executable.setOutputOptions(outputOptions);
+        executable.setConnection(getConnection(source.getUrl(), source.getUsername(), password));
+        executable.execute();
+    }
+
+    private static SchemaCrawlerOptions getOptions(Source source) {
+        setLogLevel();
+
+        SchemaCrawlerOptions options = SchemaCrawlerOptionsBuilder.newSchemaCrawlerOptions();
+        LimitOptionsBuilder limitOptionsBuilder = LimitOptionsBuilder.builder();
+
+        //TODO :  cross check this regex pattern used
+        limitOptionsBuilder.includeSchemas(new RegularExpressionInclusionRule(".*${source.getSchema}.*"))
```

Comment:
Java doesn't have string interpolation. You'll have to pull ${} out and use java string concatenation.